### PR TITLE
Build: support arm64 macOS

### PR DESCRIPTION
**Pull request by damdo**
Make the build script more flexible to support macOS M* (arm64) architectures.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -140,7 +147,7 @@ args+=(
  ${unversioned_components[@]}
 )
 
-GOOS=linux GOARCH="${arch}" GOBIN=$(pwd) ./gokr-packer "${args[@]}"
+GOOS="linux" GOARCH="${arch}" GOBIN=$(pwd) ./gokr-packer "${args[@]}"
```
Comment (andig) [inline build.sh]: linux should work without the quotes

**Comment by andig** [comment]
That should work, thank you!

**Comment by damdo** [comment]
Awesome, thanks.

**Comment by andig** [comment]
Packer install is still broken as it picks up the external arch/os. Need builder_arch too:

```
GOOS=linux GOARCH=arm64 ./build.sh
gokrazy ensuring version: 'latest'
gokr-packer version 'v0.0.0-20221008090041-c87c806562c3
v0.5.1' is not the desired one 'latest'
fetching 'latest'..
go: cannot install cross-compiled binaries when GOBIN is set
```

**Comment by andig** [comment]
The builder_os was missing:

    version="$(GOBIN=$(pwd) go version -m ./gokr-packer 2>/dev/null | grep mod | sed 's/[[:space:]]/,/g' | cut -d ',' -f4)"
    if [[ ${gokr_packer_version} != ${version} ]]; then
        echo "gokr-packer version '${version}' is not the desired one '${gokr_packer_version}'"
        echo "fetching '${gokr_packer_version}'.."
        GOBIN=$(pwd) GOOS=${builder_os} go install "${gokr_packer_base}@${gokr_packer_version}"
    fi

**Comment by andig** [comment]
And typo in builder_os:

    if [[ "builder_$os" == "darwin" ]]; then

**Comment by damdo** [comment]
Thanks for catching those, see: https://github.com/damdo/gokrazy-on-qemu/pull/2

**Pull request merged**